src/Lecture4.hs

```diff
@@ -257,7 +398,7 @@ the file doesn't have any products.
 -}
 
 calculateStats :: String -> String
-calculateStats = error "TODO"
+calculateStats = displayStats . combineRows . fromList . mapMaybe parseRow . tail . lines
```

Comment:
`fromList` is a partial function and will throw an error on an empty list. For example, `mapMaybe` can return empty list in many cases. It's better to avoid using partial functions 🙂